### ISSUE TITLE
ONNX gives much different results than Tensorflow for TF graphs with dropout layer

### ISSUE THREAD

**Issue by jim-meyer** [onnx/tensorflow-onnx#495]
I have a Tensorflow binary classification model that uses a dropout layer. When I convert it to ONNX and run the resulting model under onnxruntime I get much different class probabilities than Tensorflow for the same inputs.
Full reproducible code that demonstrates this problem can be found here: https://github.com/jim-meyer/onnx_tf_dropout_bug

Note that I created a similar model, including dropout layer, in Keras and after converting that model with onnxmltools.convert_keras() onnxruntime returned identical probabilities as Keras did for the same inputs.
This makes me think that the root cause of the Tensorflow dropout layer bug is in the tf2onnx package. Hence I'm reporting it here and not under onnxruntime.

**Comment by jim-meyer** [comment]
Closing this as I now realize that Tensorflow requires one to set the dropout rate to 0.0 when doing inference whereas Keras seems to automatically change the dropout rate implicitly when doing inference.
This "bug" only occurred when the dropout rate was left as is from training (ie. > 0.0) and does not occur when the dropout rate is set to 0.0. So this is working as designed as far as ONNX is concerned.
